### PR TITLE
[NEW] Support extra params in UserProfile

### PR DESCRIPTION
**Pull request by EugeneButusov**
Implement extra values for User Profile (like access_token).
Differencies with iOS implementation:
1. Facebook module doesn't provide us with 'expiration_date' field (because this is not provided by SimpleFacebook).
2. Google+ module doesn't provide us with 'refresh_token' and 'expiration_date' fields (because this is not provided by GMS module).

### CODE REVIEW

File: SoomlaAndroidProfile/src/com/soomla/profile/domain/UserProfile.java [hunk 1]
```diff
@@ -66,21 +87,24 @@ public UserProfile(JSONObject jsonObject) throws JSONException {
...
+        this.mAvatarLink = jsonObject.optString(PJSONConsts.UP_AVATAR, null);
+        this.mLocation = jsonObject.optString(PJSONConsts.UP_LOCATION, null);
+        this.mGender = jsonObject.optString(PJSONConsts.UP_GENDER, null);
+        this.mLanguage = jsonObject.optString(PJSONConsts.UP_LANGUAGE, null);
+        this.mBirthday = jsonObject.optString(PJSONConsts.UP_BIRTHDAY, null);
+
+        JSONObject extraJson = jsonObject.optJSONObject(PJSONConsts.UP_EXTRA);
+        if (extraJson != null) {
+            HashMap<String, Object> map = new HashMap<String, Object>();
+            while (extraJson.keys().hasNext()) {
```
Comment (vedi) [inline SoomlaAndroidProfile/src/com/soomla/profile/domain/UserProfile.java]: @EugeneButusov, replace with foreach iterator

File: social-providers/android-profile-facebook/src/com/soomla/profile/social/facebook/SoomlaFacebook.java [hunk 2]
```diff
@@ -706,9 +702,14 @@ public void call(String errorMessage) {
                     @Override
                     public void onComplete(Profile response) {
                         super.onComplete(response);
+                        HashMap<String, Object> extraDict = new HashMap<String, Object>();
+                        extraDict.put("access_token", SimpleFacebook.getInstance().getAccessToken());
+                        try {
+                            extraDict.put("permissions", new JSONArray(SimpleFacebook.getInstance().getGrantedPermissions().toArray()));
```
Comment (vedi) [inline social-providers/android-profile-facebook/src/com/soomla/profile/social/facebook/SoomlaFacebook.java]: @EugeneButusov, as I remember in iOS we have comma separated list of permissions. I think we should get the same here.

File: social-providers/android-profile-google/src/com/soomla/profile/social/google/SoomlaGooglePlus.java [hunk 3]
```diff
@@ -360,17 +366,39 @@ public void like(Activity parentActivity, String pageId) {
...
+                protected String doInBackground(Void... params) {
+                    String token = null;
+
+                    try {
+                        String SCOPES = "https://www.googleapis.com/auth/plus.login";
+                        token = GoogleAuthUtil.getToken(
+                                WeakRefParentActivity.get(),
+                                Plus.AccountApi.getAccountName(GooglePlusAPIClient),
+                                "oauth2:" + SCOPES);
+                    } catch (Exception exception) {  }
```
Comment (vedi) [inline social-providers/android-profile-google/src/com/soomla/profile/social/google/SoomlaGooglePlus.java]: @EugeneButusov, we need to elaborate this. Let's discuss.

File: social-providers/android-profile-twitter/src/com/soomla/profile/social/twitter/SoomlaTwitter.java [hunk 4]
```diff
@@ -676,10 +679,13 @@ private UserProfile createUserProfile(User user) {
                 }
             }
         }
-
-        // Twitter does not supply email access: https://dev.twitter.com/faq#26
+        HashMap<String, Object> extraDict = new HashMap<String, Object>();
+        try {
+            extraDict.put("access_token", twitter.getOAuthAccessToken().getToken());
+        } catch (Exception exc) { }
```
Comment (vedi) [inline social-providers/android-profile-twitter/src/com/soomla/profile/social/twitter/SoomlaTwitter.java]: @EugeneButusov, and this one as well.

File: social-providers/android-profile-facebook/src/com/soomla/profile/social/facebook/SoomlaFacebook.java (hunk 2 above)
Comment (EugeneButusov) [inline social-providers/android-profile-facebook/src/com/soomla/profile/social/facebook/SoomlaFacebook.java]: @vedi, no, FBAccessTokenData returns array of NSStrings and each string in this list defines a permission.
More info here: https://developers.facebook.com/docs/reference/ios/3.5/class/FBAccessTokenData/

Comment (vedi) [inline social-providers/android-profile-facebook/src/com/soomla/profile/social/facebook/SoomlaFacebook.java]: @EugeneButusov, ok, just to get sure we have the same data in iOS and Android

File: SoomlaAndroidProfile/src/com/soomla/profile/domain/UserProfile.java (hunk 1 above)
Comment (EugeneButusov) [inline SoomlaAndroidProfile/src/com/soomla/profile/domain/UserProfile.java]: @vedi, ok, I'll do that.

**Comment by vedi** [comment]
@refaelos, @borisspektor, it's ready to merge

**Pull request merged**